### DISCUSSION TITLE
Error with Prisma + Nextjs + Serverless Framework Monorepo

### DISCUSSION THREAD

**Discussion by tavindev** [prisma/prisma#11706] [Q&A] [answered]
I've been using Prisma with NextJs for a long time now, but I want to take a step further and add other Serverless solutions to my stack.
I then started learning about the Serverless Framework and created a Lerna monorepo to use Prisma with both NextJs API Routes and AWS Lambdas. However, I having a hard time making Prisma work with NextJs. 

I have three packages

```bash
├── packages
│   ├── app
│   ├── database
│   │   ├── prisma
│   │   ├── src
│   │   │   ├── index.ts
│   ├── serverless
```

And both `serverless` and `app` packages import the Prisma Client from the `prisma` package.

`prisma/src/index.ts`
```typescript
import { PrismaClient } from "@prisma/client";

export const database = new PrismaClient();
```

When I call a lambda function, Prisma works fine. On the other hand, when I call a NextJs API Route, I get an error:
```
https://nextjs.org/docs/messages/module-not-found
wait  - compiling /_error (client and server)...
error - ../database/node_modules/@prisma/client/runtime/index.js:27241:0
Module not found: Can't resolve '_http_common'

Import trace for requested module:
../database/node_modules/.prisma/client/index.js
../database/node_modules/@prisma/client/index.js
../database/dist/index.js
./src/pages/api/addUser.ts

https://nextjs.org/docs/messages/module-not-found
```

Here's the NextJs API Route
```typescript
import { NextApiRequest, NextApiResponse } from "next";

import { database } from "@p2pmarket/database";

export default async (req: NextApiRequest, res: NextApiResponse) => {
    const user = await database.user.create({});

    if (user) {
        res.status(201).json({ user });
    } else {
        res.status(400).send("Error");
    }
};
```

**Comment by tavindev** [comment, marked as answer]
Configuring lerna to use Yarn Workspaces fixed the problem.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
